### DISCUSSION TITLE
Issues with the license

### DISCUSSION THREAD

**Discussion by Atemu** [RikkaApps/Shizuku#359] [General]
The README states that:

> * For the project as a whole, it is not free.
You are **FORBIDDEN** to distribute the apk compiled by **you**
(including modified, e.g., rename app name "Shizuku" to something else)
to any store (IBNLT Google Play Store, F-Droid, Amazon Appstore etc.).

Which makes this software non-open-source/unfree/proprietary according to the [OSI's Open-Source Definition](https://opensource.org/osd/) (violates 3.). 

To avoid confusion, this part of the license text should also be put in the LICENSE file such that Github recognises the license as a custom one (which it is) rather than Apache-2.0 (which it isn't).

Shizuku being proprietary is, IMHO, a bit sad as it very likely prevents Shizuku from ever getting wide-spread adoption in the FOSS community.  
I strongly believe that any project positioning itself as a "standard API" must be Open-Source rather than proprietary and I don't think I'm alone with this opinion.

On the other hand, I also very much understand the desire to prevent copy-cats and corporate profiteering which I assume the current proprietary license is meant to prevent.

For both of these issues (and a couple others aswell) there are solutions that do not require an unfree software license however:

Using a restrictive copyleft FOSS license such as the GPLv3 instead of an unrestrictive FOSS license such as MIT or Apache-2.0 prevents anyone (including soulless corporations) from creating their own product out of your software without "giving back" to you. Anything they add to their version would have to be open aswell. See the Linux kernel's success on that for instance.  
Note that, while ordinarily Apache-2.0 is GPLv3-compatible and could simply be switched over, the term you added means that you require explicit approval of all contributors to change the license. It's best to do this early on in the project when the amount of contributors is still manageable.

The "proper" way to deal with copycats is to register a trademark for both the name and the logo. Anyone using "Shizuku" with its original name and icon would then require explicit approval from the trademark holder in order to use the name and/or icon in any "official" capacity in any place that has trademark laws.  
See for instance the Firefox trademark. Debian has famously had to ship their Firefox without the official branding as "Iceweasel" for many years until they've come to terms with Mozilla.  
A trademark also prevents other apps that aren't actually derived from Shizuku's code from posing as Shizuku because that too would violate the trademark. With the current license, an adversary could publish malware on the play store and call it "Shizuku" without any possible recourse by the project; thereby damaging the project's reputation.

Another possibility a trademark opens is the ability to grant permission to certain entities you trust (such as F-droid) to build and distribute an "official" version of Shizuku. This is not possible currently and will never be possible as long as the custom license applies to this project.  
On that note: In my lay interpretation, the wording of the current license means that *any* store listing (including the "official" ones presented on the website) are *not* allowed as there is no exception. You are likely in violation of your 3rd party contributors' copyright. They probably won't sue but, technically, they could. I'd highly recommend consulting a legal expert on the matter before it comes to that.  
I personally wouldn't be taking that risk; a license change might therefore be necessary anyways and, as I said, better do this sooner than later as the red tape involved only gets more, not less, over time.

Please note that IANAL and that this is written with best intentions. I want this project to succeed and I think what I described is necessary in order for that to happen.

**Comment by Cuteistfox** [comment]
just do this

**Reply by Frydzyslaw** [reply]
Ok

**Reply by Frydzyslaw** [reply]
> []()![proguard-rules.pro.txt](https://github.com/RikkaApps/Shizuku/files/12923095/proguard-rules.pro.txt)

**Comment by Pandapip1** [comment]
The LGPL is also the correct license for this project, from what @RikkaW has said.

**Comment by haruue** [comment]
Answered in  https://github.com/RikkaApps/Shizuku/issues/69#issuecomment-1171409393
> We don't care if the app is FOSS or not, we expose the code for the purpose of sharing knowledge. Allow anyone to use our code, include closed source software (GPL takes away the right of closed source software to use our code). However, distribution of modified apks is not welcome, which can make it difficult to handle feedback. F-Droid modifies the source code at build time, you should not trust F-Droid, GitHub Actions is a verifiable and reproducible build.

**Reply by Atemu** [reply]
> We don't care if the app is FOSS or not

I've highlighted the problem with that in my initial comment.

> we expose the code for the purpose of sharing knowledge. Allow anyone to use our code, include closed source software (GPL takes away the right of closed source software to use our code).

Sadly, the current license does not allow that. As mentioned IANAL but, in my interpretation, anyone building an APK of any sort containing any amount of Shizuku's code is not allowed to distribute it. Proprietary or not.

As others mentioned, if you want to allow corporate profiteering, the LGPL is for you.

Btw, on that note: You are technically allowed to distribute the project in any other format as the custom license only excludes APKs. If android ever changes formats for installables, it's free reign for everyone. Yet another hole to plug.

> distribution of modified apks is not welcome, which can make it difficult to handle feedback.

I can absolutely understand that. This has been a point of tension between upstream developers and downstream distros probably ever since distros were a thing. Nowadays, most distros strive to be as close to upstream as possible however.  
Unless there are significant issues introduced (i.e. new proprietary dependency), you can expect an F-droid update within a week after you release.

> F-Droid modifies the source code at build time, you should not trust F-Droid

I am not aware of any such process. What sometimes happen is that F-droid has to fork some projects to make them not use proprietary dependencies or disable other anti-features such as spyware.

If you've done everything correctly, your APK should be the exact same as the one F-droid produces (modulo signatures). F-droid is hugely invested in reproducible builds, see i.e. this recent blog post of them with some news about upstream signatures aswell: https://f-droid.org/en/2023/09/03/reproducible-builds-signing-keys-and-binary-repos.html

> GitHub Actions is a verifiable and reproducible build

The entire trust for Github Actions hinges on trusting M$; a tech giant that has shown time and time again that it is not trustworthy.

Reproducible builds are what builds trust here.

**Reply by eternal-sorrow** [reply]
>you should not trust F-Droid

Who hurt you?

**Comment by LadyDarwe** [comment]
Feel like I'm missing something. Why does the app I have from the playstore contain quarantined malicious code? 
It's the LICENSE. I read above that it's vulnerable to that. 
I'm not 100% educated on malware and such yet, but does this mean I should delete it plain and simple? 
Or no, because it's quarantined? 
Should I have my app "keep an eye on it"  instead of whitelist? 
Would FDroid positively have a malware free one? If so, I'll just switch out the app for fdroid's. 
Thanks to anyone who can help with that. I know it's pretty basic, but I have to start somewhere 👾

**Reply by Pandapip1** [reply]
The LICENSE file refers to the terms and conditions. Due to it being a non-code file, it is practically impossible for it to contain malware.

The reason it might appear suspicious to whatever antimalware you're using is because the entire reason for the app existing is for privilege escalation - which just also happens to be a tactic that a lot of malware uses.

The app is open source (although non-libre), so you can audit it yourself if you have any concerns.

**Reply by Atemu** [reply]
It is neither libre nor open source. As it currently stands, it is source-available proprietary software.

**Reply by Pandapip1** [reply]
It's open source, in that the source code is available to be inspected. It's non-free and non-libre, however.

**Reply by LadyDarwe** [reply]
Thanks! @Pandapip1 
I have a better understanding of the topic in general now.

**Reply by Pandapip1** [reply]
Heads up: What I said about open source in the above comment was wrong. I confused open source with source available. This is not open source, but is source available.

**Comment by Mammania** [comment]
Would  be great if we get viper4android installation through  shizuku

**Comment by microbearsmb** [comment]
Agreed for the most part with @Atemu here. Regardless of which direction you choose to take the license, it would be beneficial to everyone interested in Shizuku's capabilities if it was clarified. Would you consider this, @RikkaW ?

**Comment by nubesurrealista** [comment]
> Answered in  https://github.com/RikkaApps/Shizuku/issues/69#issuecomment-1171409393
> > We don't care if the app is FOSS or not, we expose the code for the purpose of sharing knowledge. Allow anyone to use our code, include closed source software (GPL takes away the right of closed source software to use our code). However, distribution of modified apks is not welcome, which can make it difficult to handle feedback. F-Droid modifies the source code at build time, you should not trust F-Droid, GitHub Actions is a verifiable and reproducible build.
> 
> 

The problem with that position is that someday someone will reverse-engineer this code or manage to replicate what shizuku does and the community is not going to be on your side. If you want to exclude yourselves from the open-source community, I guess you can. But when this project needs help or you get into a fight with someone for "stealing/copying" your code ... You will probably be the ones who lose :)

I am a Shizuku user and I will continue to use it, but, as an open-source advocate, I hope someone will come and tear that license to pieces.

**Comment by boreals-back-again** [comment]
> Answered in  https://github.com/RikkaApps/Shizuku/issues/69#issuecomment-1171409393
> > We don't care if the app is FOSS or not, we expose the code for the purpose of sharing knowledge. Allow anyone to use our code, include closed source software (GPL takes away the right of closed source software to use our code). However, distribution of modified apks is not welcome, which can make it difficult to handle feedback. F-Droid modifies the source code at build time, you should not trust F-Droid, GitHub Actions is a verifiable and reproducible build.

If anyone can "use" your code, what stops anyone from developing an alternative to Shizuku that respects the user?
